knowledge-base/grid-filterrow-bool-customization.md

```diff
@@ -0,0 +1,150 @@
+    <GridColumns>
+        <GridColumn Field=@nameof(Employee.Name) />
+        <GridColumn Field=@nameof(Employee.IsOnLeave) Title="On Vacation">
+            <FilterCellTemplate>
+
+                <!-- Dropdown list for selecting filter options -->
+                <TelerikDropDownList Data="@FilterOptions" Value="@DropDownListValue"
+                                     ValueChanged="@((string newValue) => ApplyFilter(newValue, context))">
+                    <DropDownListSettings>
+                        <DropDownButtonPopupSettings Height="auto" />
+                    </DropDownListSettings>
+                </TelerikDropDownList>
+
+                <!-- Button to clear the filter -->
+                <TelerikButton ButtonType="ButtonType.Button"
+                               Class="k-clear-button-visible ml-2"
+                               Icon="@SvgIcon.FilterClear"
+                               Enabled="@(DropDownListValue != "All")"
+                               OnClick="@(async () =>
+                                          {
+                                              await context.ClearFilterAsync();
+                                              DropDownListValue = "All";
+                                          })">
+                </TelerikButton>
+            </FilterCellTemplate>
+        </GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Employee> GridData { get; set; } = new();
+
+    // Custom options for the dropdown filter
+    private List<string> FilterOptions { get; set; } = new() { "All", "Yes", "No", "No Info" };
+
+    // Selected value in the dropdown filter
+    private string DropDownListValue { get; set; } = "All";
+
+    // Method to apply the filter based on the selected value in the dropdown
+    private async Task ApplyFilter(string newValue, FilterCellTemplateContext context)
+    {
+        DropDownListValue = newValue;
+
+        var filter = context.FilterDescriptor.FilterDescriptors[0] as FilterDescriptor;
+        if (filter != null)
+        {
+            switch (DropDownListValue)
+            {
+                case "Yes":
+                    filter.Value = true;
+                    filter.Operator = FilterOperator.IsEqualTo;
+                    break;
+                case "No":
+                    filter.Value = false;
+                    filter.Operator = FilterOperator.IsEqualTo;
+                    break;
+                case "No Info":
+                    filter.Value = null;
+                    filter.Operator = FilterOperator.IsNull;
+                    break;
+                case "All":
+                    await context.ClearFilterAsync();
+                    return;
+            }
+
+            // Apply the filter
+            await context.FilterAsync();
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        for (int i = 0; i < 30; i++)
```

Comment:
Do not use zero as an ID, because this is the default value of the numeric types.
```suggestion
        for (int i = 1; i <= 30; i++)
```